core/rint/src/TRint.cxx

```diff
@@ -49,6 +49,47 @@
 #include <signal.h>
 #endif
 
+#ifndef R__WIN32
+# ifndef R__MACOSX
+#  include <dirent.h>
+# endif
+#endif
+
+namespace {
+   /// Should IMT be automatically enabled?
+   /// Yes, unless the machine is (likely) shared by multiple users,
+   /// e.g. batch / grid or lxplus. None of them have X running, so
+   /// use that as criterium.
+   static bool ShouldAutomaticallyEnableIMT()
+   {
+#ifdef R__WIN32
+      // Windows: assume single user.
+      return true;
+#elif R__MACOSX
+      return true;
+#else
+   auto hasX = []() {
+      // All current X-flavors signal their presence through /tmp/.X11-unix/
+      DIR *dir = opendir("/tmp/.X11-unix/");
```

Comment:
Is the existence of this directory really a requirement for X11 to work? On my CC7 desktop it currently does not exist, and so far I did not observe any issues with X.

The directory seems to be created at boot time, but is removed when it is older than 10 days. Probably this means 10 days after the X server has been restarted and the unix socket has been created in this directory. By only locking my screens over night/weekend, and not logging off and on again, I apparently reach this limit.

Reply:
Thanks! I see that there is an SElinux workaround. To test whether that's active for you, could you check whether `socat ABSTRACT-CONNECT:/tmp/.X11-unix/X0 -` says "Connection refused"? (I bet it doesn't.)

Reply:
The `socat` command is indeed working. In addition `netstat -an` prints
```
unix  2      [ ACC ]     STREAM     LISTENING     82470    /tmp/.X11-unix/X0
unix  2      [ ACC ]     STREAM     LISTENING     82469    @/tmp/.X11-unix/X0
unix  3      [ ]         STREAM     CONNECTED     94741    @/tmp/.X11-unix/X0
```
and several lines like the last one all connected to the socket in the abstract namespace. None of the X clients is connected to the 'normal' socket.